tap_github/repository_streams.py

```diff
@@ -1916,6 +1921,808 @@ def query(self) -> str:
     ).to_dict()
 
 
+class DiscussionCategoriesStream(GitHubGraphqlStream):
+    """Defines stream fetching discussions categories from each repository."""
+
+    name = "discussion_categories"
+    query_jsonpath = "$.data.repository.discussionCategories.nodes.[*]"
+    primary_keys: ClassVar[list[str]] = [
+        "node_id"
+    ]  # id:databaseId is not available for the categories object
+    replication_key = "updated_at"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys: ClassVar[list[str]] = ["repo_id"]
+    ignore_parent_replication_key = True  # Repository's updated_at does not change when a new discussion category is added/modified  # noqa: E501
+    use_fake_since_parameter = True
```

Comment:
Is this intended to extract incrementally or full extract every run?  I think we won't want this setting if it's the latter.